src/functions-reference/higher-order_functions.Rmd

```diff
@@ -141,6 +142,138 @@ Stan's automatic differentiation.
 
 ## Ordinary Differential Equation (ODE) Solvers {#functions-ode-solver}
 
+Stan provides built-in ordinary differential equation (ODE) solvers.
+Because the first argument to each of the solvers is a function itself,
+the ODE solvers in Stan are considered high order functions.
+
+### Specifying an Ordinary Differential Equation as a Function
+
+A system of ODEs is specified as an ordinary function in Stan within
+the functions block. The ODE system function must have the same return
+type and first two arguments as the following function:
+
+```
+ vector ode(real time, vector state, ...)
+```
+
+`...` are any number of arguments passed through unmodified from the ODE integrate
+call.
+
+The ODE system function should return the derivative of the state with
+respect to time at the time and state provided. The length of the returned
+vector must match the length of the state input into the function.
+
+The arguments to this function are:
+
+*   *`time`*, the time to evaluate the ODE system
+
+*   *`state`*, the state of the ODE system at the time specified
+
+*   *...*, all other arguments passed through from the integrate call. Each
+argument can be any type.
```

Comment:
Before this section I would show an example signature for the ODE integrators because

> all other arguments passed through from the integrate call

Doesn't really make sense without knowing what the integrators look like

```suggestion

Where the arguments to the function are:

*   *`time`*, the time to evaluate the ODE system

*   *`state`*, the state of the ODE system at the time specified

*   *...*, Any additional arguments needed in the ODE. Each of these can be arguments of any type.

The ODE system function should return the derivative of the `state` with
respect to `time` at the `time` and `state` provided. The length of the returned
vector must match the length of the state input into the function.
```

Reply:
I moved the ODE system function definition below the signatures and the arguments to the integrator functions.

It's still slightly confusing because the first argument to the integrator function is the ODE system function, but there's a little inline doc there that hopefully works.